cognite_toolkit/cdf_tk/load.py

```diff
@@ -60,249 +74,119 @@ def __next__(self):
+    """
 
-def load_timeseries_metadata(
-    ToolGlobals: CDFToolConfig,
-    file: str,
-    drop: bool,
-    dry_run: bool = False,
-    directory=None,
-) -> None:
-    if directory is None:
-        raise ValueError("directory must be specified")
-    client = ToolGlobals.verify_client(capabilities={"timeSeriesAcl": ["READ", "WRITE"]})
-    files = []
-    if file:
-        # Only load the supplied filename.
-        files.append(file)
-    else:
-        # Pick up all the .yaml files in the data folder.
-        for _, _, filenames in os.walk(directory):
-            for f in filenames:
-                if ".yaml" in f:
-                    files.append(f)
-    # Read timeseries metadata
-    timeseries = TimeSeriesList([])
-    for f in files:
-        timeseries.extend(
-            TimeSeriesList.load(
-                load_yaml_inject_variables(Path(f"{directory}/{f}"), ToolGlobals.environment_variables())
-            ),
-        )
-    if len(timeseries) == 0:
-        return
-    print(f"[bold]Uploading {len(timeseries)} timeseries to CDF...[/]")
-    drop_ts: list[str] = []
-    for t in timeseries:
-        # Set the context info for this CDF project
-        t.data_set_id = ToolGlobals.data_set_id
-        if drop:
-            drop_ts.append(t.external_id)
-    try:
-        if drop:
-            if not dry_run:
-                client.time_series.delete(external_id=drop_ts, ignore_unknown_ids=True)
-                print(f"  Deleted {len(drop_ts)} timeseries.")
-            else:
-                print(f"  Would have deleted {len(drop_ts)} timeseries.")
-    except Exception:
-        print(f"[bold red]ERROR:[/] Failed to delete {t.external_id}. It may not exist.")
-    try:
-        if not dry_run:
-            client.time_series.create(timeseries)
+    load_files_individually: bool = False
+    filetypes = frozenset({"yaml", "yml"})
+    api_name: str
+    folder_name: str
+    resource_cls: type[CogniteResource]
+    list_cls: type[CogniteResourceList]
+
+    def __init__(self, client: CogniteClient):
+        self.client = client
+        self.api_class = self._get_api_class(client, self.api_name)
+
+    @staticmethod
+    def _get_api_class(client, api_name: str):
+        parent = client
+        if (dot_count := Counter(api_name)["."]) == 1:
+            parent_name, api_name = api_name.split(".")
+            parent = getattr(client, parent_name)
+        elif dot_count == 0:
+            pass
         else:
-            print(f"  Would have created {len(timeseries)} timeseries.")
-    except Exception as e:
-        print("[bold red]ERROR:[/] Failed to upload timeseries.")
-        print(e)
-        ToolGlobals.failed = True
-        return
-    print(f"  Created {len(timeseries)} timeseries from {len(files)} files.")
+            raise ValueError(f"Invalid api_name {api_name}.")
+        return getattr(parent, api_name)
+
+    @classmethod
+    def create_loader(cls, ToolGlobals: CDFToolConfig):
```

Comment:
Are there any reasons for why we shouldn't push verify_client() and verify_dataset() (and then verify_space) into CogniteClient? We have the standardised env variable loading that also auth verify relies on, but that's a create_from_environment() cls method, right? I was wondering if it's worth doing that now as part of this refactoring. Having CDFToolConfig as another abstraction layer complicates understanding the code.

Reply:
Agreed, I have just not prioritized it yet.